panic: invalid pad in intro_my with perl -MO=... and use v5.39*

**Description**

With -DDEBUGGING:
```
$ ./perl -Ilib -MO=Concise -e 'use v5.39'
panic: invalid pad in intro_my: 0x55ef8a17ca20[0x55ef8a1945b0] at -e line 1
```

Also
```
$ ./perl -Ilib -MO=Deparse -e 'use v5.39'
panic: invalid pad in intro_my: 0x55cd471baa20[0x55cd471d2630] at -e line 1.
sub BEGIN {
    require v5.39;
    ()
}
```

Note this doesn't happen with anything `use v5.38` or less, only 5.39 and higher.

According to a bisect this appears to have been introduced here:
```
HEAD is now at 6d65c70ff0 Import from `builtin` version bundles should be idempotent on existing symbols
good - zero exit from ./perl -Ilib -MO=Concise -e use v5.39;
48d382667948f6a16c732c47791ca4be891a9bdb is the first bad commit
commit 48d382667948f6a16c732c47791ca4be891a9bdb
Author: Paul "LeoNerd" Evans <leonerd@leonerd.org.uk>
Date:   Fri Jan 19 15:00:35 2024 +0000

    Have `use VERSION` and `perl -E` import a corresponding `use builtin` bundle
```

More excitement: order matters:

```
leo@hevva:~/src/bleadperl/perl [git]
$ ./perl -Ilib -e 'use v5.39; use O q(Concise);'
3  <@> leave[1 ref] vKP/REFC ->(end)
1     <0> enter v ->2
2     <0> stub v ->3
-     <;> ex-nextstate(main 1 -e:1) v:us,*,&,{,$,fea=7 ->3
-e syntax OK

$ ./perl -Ilib -e 'use O q(Concise); use v5.39'
panic: invalid pad in intro_my: 0x5622c15488f0[0x5622c15689a8] at -e line 1.
```

```
tony@venus:.../git/perl6$ ./perl -Ilib -MB -e 'BEGIN { B::save_BEGINs; } use v5.39'
panic: invalid pad in intro_my: 0x55c071135a20[0x55c07114cf00] at -e line 1.
```
though I still don't know why.

--- a/builtin.c
+++ b/builtin.c
@@ -48,8 +48,9 @@ Perl_prepare_export_lexical(pTHX)
     /* We need to have PL_comppad / PL_curpad set correctly for lexical importing */
     ENTER;
     SAVESPTR(PL_comppad_name); PL_comppad_name = PadlistNAMES(CvPADLIST(PL_compcv));
-    SAVESPTR(PL_comppad);      PL_comppad      = PadlistARRAY(CvPADLIST(PL_compcv))[1];
-    SAVESPTR(PL_curpad);       PL_curpad       = PadARRAY(PL_comppad);
+    SAVECOMPPAD();
+    PL_comppad      = PadlistARRAY(CvPADLIST(PL_compcv))[1];
+    PL_curpad       = PadARRAY(PL_comppad);
 }
 
 #define export_lexical(name, sv)  S_export_lexical(aTHX_ name, sv)

--- a/lib/builtin.t
+++ b/lib/builtin.t
@@ -621,6 +621,15 @@ TODO: {
     }
 }
 
+# github #21981
+{
+    fresh_perl_is(<<'EOS', "", {}, "github 21981: panic in intro_my");
+use B;
+BEGIN { B::save_BEGINs; }
+use v5.39;
+EOS
+}
+
 # vim: tabstop=4 shiftwidth=4 expandtab autoindent softtabstop=4
 
 done_testing();

